pkg/common/providers/moaprovider/cluster.go

```diff
@@ -16,6 +17,47 @@ import (
 	"github.com/spf13/viper"
 )
 
+// IsValidClusterName validates the clustername prior to proceeding with it
+// in launching a cluster.
+func (m *MOAProvider) IsValidClusterName(clusterName string) (bool, error) {
+	// Create a context:
+	ctx := context.Background()
+
+	collection := m.ocmProvider.GetConnection().ClustersMgmt().V1().Clusters()
+
+	// Retrieve the list of clusters using pages of ten items, till we get a page that has less
+	// items than requests, as that marks the end of the collection:
+	size := 10
+	page := 1
+	searchPhrase := fmt.Sprintf("name like '%s'", clusterName)
```

Comment:
Is there a better way to look for an existing name? Instead of a `like` style lookup could we look for an exact match?

Doing that would remove the need for a loop and any ambiguity; if there's a result, the name's taken.

Reply:
This was a mistake, copied from testing ocmprovider before I fine tuned the syntax to an exact. Resolved in: e819298f